packages/block-editor/src/components/block-list/use-block-props/index.js

```diff
@@ -102,6 +103,8 @@ export function useBlockProps( props = {}, { __unstableIsHtml } = {} ) {
 	const mergedRefs = useMergeRefs( [
 		props.ref,
 		useFocusFirstElement( clientId ),
+		// Must happen after focus because we check for focus in the block.
+		useScrollIntoView( clientId ),
```

Comment:
Ideally we have a hook that does combines this: if a selected block has no focus, either focus the first element (if so desired) or scroll the block into view.

Reply:
(If we do set focus, the browser will automatically scroll it into view if needed.)